core/templates/domain/question/QuestionObjectFactory.ts

```diff
@@ -133,17 +136,18 @@ export class Question {
     return null;
   }
 
-  getUnaddressedMisconceptionNames(misconceptionsBySkill: {}): string[] {
+  getUnaddressedMisconceptionNames(
+      misconceptionsBySkill: MisconceptionSkillMap = {}): string[] {
     var answerGroups = this._stateData.interaction.answerGroups;
-    var taggedSkillMisconceptionIds = {};
+    var taggedSkillMisconceptionIds: { [stateName: string]: boolean } = {};
     for (var i = 0; i < answerGroups.length; i++) {
       if (!answerGroups[i].outcome.labelledAsCorrect &&
           answerGroups[i].taggedSkillMisconceptionId !== null) {
         taggedSkillMisconceptionIds[
-          answerGroups[i].taggedSkillMisconceptionId] = true;
+          answerGroups[i].taggedSkillMisconceptionId || 'null'] = true;
```

Comment:
Why this change?

Reply:
> 
Here because taggedSkillMisconceptionId can be null but here is is being used as an index which cannot be null, that's why it expects it to pnly take string values.
![image](https://user-images.githubusercontent.com/42764756/143374199-a26b572f-104d-4ef0-96c1-c720e45723f1.png)

Reply:
But then we shouldn't add it to the dictionary, right?

Reply:
Hmm... yes will check this. Thanks !

Reply:
So @vojtechjelinek , we are actually not adding it to the dictionary in that case 
`if (!answerGroups[i].outcome.labelledAsCorrect &&
          answerGroups[i].taggedSkillMisconceptionId !== null) {
        taggedSkillMisconceptionIds[
          String(answerGroups[i].taggedSkillMisconceptionId) ] = true;` but it's required just to deal with Typescript error.